core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -54,6 +57,8 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
   import org.apache.kafka.clients.consumer.KafkaConsumer
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  val offsets = new HashMap[TopicPartition, Long]()
```

Comment:
Cached partitions from which messages are read in case of creating too many unnecessary objects to burden the young GC.